src/witan/workspace/protocols.clj

```diff
@@ -2,3 +2,18 @@
 
 (defprotocol SendMessage
   (send-message! [this topic message]))
+
+;;;;;;;;;
+
+(defprotocol Database
+  (drop-table!
+    [this table])
+  (create-table!
+    [this table columns])
+  (insert!
+    [this table row]
+    [this table row args])
+  (select*
```

Comment:
It's more to appease the component (as in stuartsierra's) isolation and also gives us the ability to use those replacer calls, mentioned above.